### PR TITLE
[GPU] Support output idx for get_output_layout to get input_layout in multiple outputs case of reshape

### PR DESCRIPTION
**Pull request by steve-y**
### Details:
 - *Use proper output idx in get_output_layout instead using default value of 0*

### Tickets:
 - *125532*

### CODE REVIEW

**Review by kelvinchoi-intel — APPROVED** [review]
LGTM

**Comment by wilson-seok** [comment]
@steve-y  Could you make a func test?

**Comment by wilson-seok** [comment]
@steve-y And please run CI regression check with all_ir option.

File: src/plugins/intel_gpu/src/graph/reshape.cpp [hunk 1]
```diff
@@ -164,7 +164,7 @@ std::string reshape_inst::to_string(reshape_node const& node) {
 
 reshape_inst::typed_primitive_inst(network& network, reshape_node const& node) :
         parent(network, node, (!node.can_be_optimized() && node.get_output_layout().is_static()) ? true : false) {
-    auto input_layout = node.input().get_output_layout();
+    auto input_layout = node.input().get_output_layout(true, node.get_dependency_with_port(0).second);
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/graph/reshape.cpp:167]: I think `node.get_input_layout()` can be used instead

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/reshape.cpp:167]: Though `get_input_layout` impl may need similar fix. Looks like `get_input_layouts()` respects out port, but `get_input_layout` doesn't.

**Comment by vladimir-paramuzov** [comment]
Implemented in https://github.com/openvinotoolkit/openvino/pull/22417

**Pull request closed without merging**